catalog/app/components/Preview/renderers/Ngl.tsx

```diff
@@ -0,0 +1,61 @@
+import * as NGL from 'ngl'
+import * as React from 'react'
+import * as M from '@material-ui/core'
+
+async function renderNgl(blob: Blob, wrapperEl: HTMLDivElement, t: M.Theme) {
+  const stage = new NGL.Stage(wrapperEl, { backgroundColor: t.palette.common.white })
+
+  const resizeObserver = new window.ResizeObserver(() => stage.handleResize())
+  resizeObserver.observe(wrapperEl)
+
+  await stage.loadFile(blob, {
+    defaultRepresentation: true,
+    ext: 'pdb',
+  })
+  return stage
+}
+
+const useStyles = M.makeStyles((t) => ({
+  root: {
+    height: t.spacing(50),
+    overflow: 'auto',
+    resize: 'vertical',
+    width: '100%',
+  },
+}))
+
+interface NglProps extends React.HTMLAttributes<HTMLDivElement> {
+  blob: Blob
+}
+
+function Ngl({ blob, ...props }: NglProps) {
+  const classes = useStyles()
+
+  const t = M.useTheme()
+  const viewport = React.useRef<HTMLDivElement | null>(null)
+  const stageRef = React.useRef<NGL.Stage | null>(null)
+
+  const handleWheel = React.useCallback(
+    (event) => {
+      if (viewport.current?.contains(event.target)) {
+        event.preventDefault()
+      }
+    },
+    [viewport],
+  )
+
+  React.useEffect(() => {
+    if (viewport.current) {
+      renderNgl(blob, viewport.current, t).then((stage) => (stageRef.current = stage))
```

Comment:
since this reference is only used in the cleanup function, you don't need a component-level reference here (it may even cause bugs, since it's a mutable shared object), just a `let` variable would be enough, e.g.

```js
let stage
renderNgl(blob, viewport.current, t).then((s) => { stage = s })
...
return () => {
      stage?.dispose()
}
```